File: server/src/main/java/com/hyperlogix/server/domain/Stop.java
package com.hyperlogix.server.domain;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Stop {
  Node node;
  LocalDateTime arrivalTime;
  boolean arrived;

  public Stop(Node node, LocalDateTime arrivalTime) {
    this.node = node;
    this.arrivalTime = arrivalTime;
    this.arrived = false;
  }
  

}

File: client/src/domain/Stop.ts
import type { Node } from './Node' // Or a more specific Stop related Node information

export interface Stop {
  // Example properties, adjust based on actual Stop.java definition
  node: Node // Reference to the node being stopped at
  arrivalTime?: string // ISO date-time string
  arrived: boolean // Indicates if the stop has been completed
}
